clang/lib/CIR/CodeGen/CIRGenBuilder.h

```diff
@@ -266,6 +266,10 @@ class CIRGenBuilderTy : public mlir::OpBuilder {
   mlir::cir::IntType getUInt32Ty() { return typeCache.UInt32Ty; }
   mlir::cir::IntType getUInt64Ty() { return typeCache.UInt64Ty; }
 
+  mlir::cir::IntType getCustomIntTy(unsigned size, bool isSigned) {
+    return mlir::cir::IntType::get(getContext(), size, isSigned);
```

Comment:
Do you want to get rid of `typeCache` there?

Reply:
I completely forgot about the `typeCache`. Just re-writing in terms of `getUIntNTy` and `getSIntNTy` sounds good!

Reply:
Still a little bit confused) well, the `getUIntNTy` and `getSIntNTy`  returns type only for the the next sizes: 8, 16, 32, 64. Otherwise they **fail**. That's why I added this helper - to get a custom type for a bitfield of an arbitary size. So what do you mean by 
> Just re-writing in terms of `getUIntNTy` and `getSIntNTy` sounds good!

Reply:
Oh I see now, we don't support arbitrary sized types in CIR, for now only 8, 16, 32, 64. Not saying we shouldn't, but that needs to be a plan of its own, with some more rationale and perhaps intended use cases. 

For now you can round up to the next size between 8, 16, 32, 64 and implement in those terms, or even use a fixed type (u32i). In case you need to adjust the math a bit because of that, it's also fine.